pages/common/ncc.md

```diff
@@ -0,0 +1,29 @@
+# ncc
+
+> Compile a Node.js application into a single file.
+> Supports TypeScript, binary addons and dynamic requires.
```

Comment:
`dynamic requires` sounds a litttle strange. Should this mean `dynamic requirements` or is this correct?

Reply:
In Node.js (only, not browser JavaScript :/), there's a function called `require()`, and it's used to import classes, functions, variables, and any other value from other `.js` files (exported via the JavaScript object `module.exports`).  A sample would look like this:

- `hello.js`

  ```js
  function hello() {
    return "Hello, World!";
  }
  module.exports = hello;
  ```

- `example.js`

  ```js
  const hello = require("./hello"); // in Node.js, the .js extension is implied and usually omitted
  console.log(hello());
  ```

- Running:

  ```console
  $ node example.js
  Hello, World!
  ```

"dynamic require" here is referring to the fact that, since `require` is just a normal function, it can be used dynamically. Use cases include conditionally `require()`ing a module, looping over a list of files and `require()`ing them (e.g. to load plugins, or load configuration JSON files (because `require()` supports `.json` files too)).  You can also pass variables to it, not just hardcoded strings.

Reply:
We could possibly use `... dynamic require()s.`? Although I don't love that.

Reply:
Thanks for the explaination. I'd say it's fine like this.